File: sample/src/main/java/com/mateuszkoslacz/moviper/sample/viper/contract/FullscreenPhotoContract.java
package com.mateuszkoslacz.moviper.sample.viper.contract;

import android.app.Activity;

import com.hannesdorfmann.mosby.mvp.MvpPresenter;
import com.hannesdorfmann.mosby.mvp.MvpView;
import com.mateuszkoslacz.moviper.iface.interactor.ViperInteractor;
import com.mateuszkoslacz.moviper.iface.presenter.interactor.ViperPresenterForInteractor;
import com.mateuszkoslacz.moviper.iface.presenter.routing.ViperPresenterForRouting;
import com.mateuszkoslacz.moviper.iface.routing.ViperRouting;

public interface FullscreenPhotoContract {

    interface Presenter extends MvpPresenter<View> {
        // Defines what methods the View can invoke on the Presenter.
        // In most cases there will be user interactions and View lifecycle events.

        void onViewCreated();
    }

    interface View extends MvpView {
        // Defines what methods the Presenter can invoke on the View
        // In most cases there will be manipulating ui and displaying data or errors.

        void setPhoto(String photoUrl);
    }

    interface Interactor extends ViperInteractor<PresenterForInteractor> {
        // Defines what methods the Presenter can invoke on the Interactor.
        // In most cases there will be data saving and querying.

    }

    interface PresenterForInteractor extends ViperPresenterForInteractor {
        // Defines what methods the Interactor could invoke on the Presenter.
        // In most cases there will be data received callbacks and error notifying.

    }

    interface Routing extends ViperRouting<Activity, PresenterForRouting> {
        // Defines what methods the Presenter can invoke on the Routing.
        // In most cases there will be starting another activities, services and using system
        // framework, ie. scheduling alarms or sending broadcasts.
        // In the case of a fragment being the view, there also will be manipulating
        // the root Activity, ie. switching fragments.

    }

    interface PresenterForRouting extends ViperPresenterForRouting {
        // Defines what methods the Routing can invoke on the Presenter.
        // In most cases there will be system framework interaction callbacks and error notifying.

    }

}

File: sample-super-rx-ai-kotlin/src/main/java/com/mateuszkoslacz/moviper/rxsample/viper/contract/FullscreenPhotoContract.kt
package com.mateuszkoslacz.moviper.rxsample.viper.contract

import android.app.Activity
import com.hannesdorfmann.mosby3.mvp.MvpPresenter
import com.hannesdorfmann.mosby3.mvp.MvpView
import com.mateuszkoslacz.moviper.iface.interactor.ViperRxInteractor
import com.mateuszkoslacz.moviper.iface.routing.ViperRxRouting

interface FullscreenPhotoContract {

    interface Presenter : MvpPresenter<View>

    interface View : MvpView {
        fun setPhoto(photoUrl: String)
    }

    interface Interactor : ViperRxInteractor

    interface Routing : ViperRxRouting<Activity>

}
